w06-ruby-on-rails/d4-dawn-intro-rails/README.md

```diff
@@ -0,0 +1,195 @@
+    def index
+    end
+end
+```
+
+Wow another error! `*Missing template welcome/index...*` Since we have a `welcome` controller and an `index` method, Rails automatically will try to render a view with the path `app/views/welcome/index.html.erb`.  A directory `/welcome` should already exist, it was generated when the welcome controller was generated (thanks, Rails!). Inside of there add the file `index.html.erb` and inside of the file add some html:
+
+app/views/welcome/index.html.erb
+
+```html
+<h1>I make internets with Rails</h1>
+<img src="http://i.giphy.com/SPZFhfUJjsJO0.gif" alt="learning internet" style="width: 300px">
+```
+Visit your root route one more time.
+
+Nice!
+
+## View (5m)
+
+- This will not change the behavior, but you may wish to be more explicit in our controller by stating to `render` the template `index`
+
+`app/controllers/welcome_controller.rb`
+
+```ruby
+def index
+  render('index')
+end
+```
+
+Could we change these names to whatever we want?
+Absolutely!
+
+But don't. Conventions are established for a reason. :wink:
+
+## ERB (10m)
+
+`.html.erb` files are templates that are processed with embedded ruby, `.erb` to generate an `.html` file. This is known as **server-side templating**. This enables Rails to serve up dynamic views based on the data it is served. It is very similar to EJS, the view templating that we used in our Express apps.
+
+Let's say we want to pass a random number to our view from 0-100... Try adding this to your html:
+
+```html
+<h1>I make internets with Rails</h1>
+<p>Random number is... <%= Random.new.rand(100) %></p>
+<img src="http://i.giphy.com/SPZFhfUJjsJO0.gif" alt="learning internet" style="width: 300px">
+```
+
+Woah, what is happening? Ruby is being evaluated first and the result is printed into our html. the `<%` symbols escape our html. Remember from previous lessons we've seen on templating `<%` will *evaluate* the code while `<%=` will *interpolate* the result.
+
+## Passing Data to our View (10m)
+
+There's certainly some business logic happening in our View. This is bad. Our view should only be concerned with presenting the data, but not actually generating it, that is a violation of **separation of concerns**. To fix this, let's move the `Random.new.rand(100)` code to our controller and set that equal to a variable we will pass into our view.
+
+In Express this might have looked like:
+
+```javascript
+app.get(function(req, res) {
+  res.render('index', { random: Math.random() })
+  })
+```
+
+But in Rails it looks like this:
+
+`app/controllers/welcome_controller.rb`
+
+```ruby
+class WelcomeController < ApplicationController
+  def index
+    @random = Random.new.rand(100)
+    render('index')
+  end
+end
+```
+
+Notice we did not create a variable named `random` instead we created an instance variable named `@random`, the **@** is VERY important. Normal (local) variables' scope do not reach the view, only **instance variables**' scope reach the view.
```

Comment:
Notice we did not create a **local variable** named `random` instead we created an instance variable named `@random`, the **@** is VERY important. Normal (local) variables' scope do not reach the view, only **instance variables**' scope reaches the view.

Honestly the plurality of the last sentences still feels weird.  I think maybe 'scopes' should be plural.  Possible alternative wording:  
The scope of local variables does not reach into the view.  Only controller **instance variables** are available (in-scope) in the view.